File: src/main/java/me/anonymoussoftware/vacancymanager/model/Area.java
package me.anonymoussoftware.vacancymanager.model;

import org.json.JSONObject;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Area {

    private final int id;
    private final String name;

    public static Area fromJson(JSONObject json) {
        return new Area(json.getInt("id"), json.getString("name"));
    }

    public JSONObject toJson() {
        JSONObject result = new JSONObject();
        result.put("id", this.id);
        result.put("name", this.name);
        return result;
    }
}

File: src/main/kotlin/me/anonymoussoftware/vacancymanager/model/Area.kt
package me.anonymoussoftware.vacancymanager.model

import org.json.JSONObject

class Area (override val id: Int,
            override val name: String) : ModelWithNameAndId {

    fun toJson(): JSONObject {
        val result = JSONObject()
        result.put("id", this.id)
        result.put("name", this.name)
        return result
    }

    companion object : JsonDeserializer<Area> {
        override fun fromJson(json: JSONObject) : Area =
            Area(json.getInt("id"), json.getString("name"))
    }
}
